src/Umbraco.Core/Models/PublishedContent/VariationContext.cs

```diff
@@ -30,4 +30,23 @@ public VariationContext(string? culture = null, string? segment = null)
     /// <param name="contentId"></param>
     /// <returns></returns>
     public virtual string GetSegment(int contentId) => Segment;
+
+    /// <summary>
+    /// A copy of the VariationContext with a different culture.
+    /// </summary>
+    /// <param name="culture">The new culture.</param>
+    /// <remarks>Does not modify the original VariationContext.</remarks>
+    /// <returns>A copy of the VariationContext (same segment) with the culture from <paramref name="culture"/>.</returns>
+    public VariationContext WithCulture(string? culture)
```

Comment:
Please note that this creates a new `VariationContext` instance, but this class can be inherited (in case you need a custom `GetSegment()` implementation) and will therefore revert back to the default implementation when calling this or the `WithSegment()` method...

Maybe a better approach is to add a protected setter to the `Culture` and `Segment` properties and change these methods to `SetCulture`/`SetSegment` (potentially also make them virtual, in case a custom implementation needs to validate the values - which can currently already be done in the constructor)?

Reply:
Good point. I was creating to keep things immutable. Alternatively `WithSegment` could be `virtual`.

Should I amend as per your suggestion?